src/containers/TemplateBuilder/template/General/Filters.tsx

```diff
@@ -112,14 +116,36 @@ const Filters: React.FC = () => {
     }
   }
 
+  const addDashboardRestriction = async () => {
+    const currentRestrictions = template.dashboardRestrictions ?? []
+    if (selectedRestriction)
+      updateTemplate(template.id, {
+        dashboardRestrictions: [...currentRestrictions, selectedRestriction?.code],
+      })
+    setSelectedRestriction(undefined)
+    setSelectedRestrictFilter(undefined)
+  }
+
+  const removeDashboardRestriction = async () => {
+    const newRestrictions = (template.dashboardRestrictions ?? []).filter(
+      (el) => el !== selectedRestrictFilter
+    )
+    setSelectedRestrictFilter(undefined)
+    setSelectedRestriction(undefined)
+    updateTemplate(template.id, {
+      dashboardRestrictions: newRestrictions.length > 0 ? newRestrictions : null,
+    })
+  }
+
   return (
     <>
       <Header as="h3">Dashboard Filters</Header>
       <div className="flex-row-start-center">
         <DropdownIO
           value={selectedFilter.id}
+          placeholder="Select filter"
           title="Filters"
-          options={allFilters}
+          options={[...allFilters, newFilter].filter((fil) => !filterJoinCodes.includes(fil?.code))}
```

Comment:
Not strictly part of this issue, but I thought it was better if filters that had already been selected no longer appeared in the menu.